rules/prefer-starts-ends-with.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const doesNotContain = (string, chars) => chars.every(char => !string.includes(char));
+
+const isSimpleString = string => doesNotContain(
+	string,
+	['^', '$', '+', '[', '{', '(', '\\', '.', '?', '*']
```

Comment:
Sorry, totally forgot to answer.

Yes, we're on the same page.

I really don't know how to work with this.
I think we could ask help from @firasdi, who made the (awesome) https://regex101.com/. @firasdi, would you know how to tell whether a string is a normal string or has symbols (with the issue of escaping/unescaping) that makes it a regex that does more than just `.contain()`? If you know a library that does this, that'd be even better :sweat_smile: 

In case this gets too hard, we could just merge the PR without the regex part and handle that later.